Modules/IO/NIFTI/include/itkNiftiImageIO.h

```diff
@@ -28,28 +28,53 @@
 namespace itk
 {
 
-
-/** \class Analyze75Flavor
+/**\class NiftiImageIOEnums
+ * \brief
  * \ingroup ITKIONIFTI
- * Enum used to define way to treat legacy Analyze75 files
  */
-enum class Analyze75Flavor : uint8_t
+class NiftiImageIOEnums
 {
-  /** Behavior introduced in ITK4.0 by NIFTI reader interpreting Analyze files */
-  AnalyzeITK4 = 4,
-  /** Will ignore orientation code and negative pixel dimensions */
-  AnalyzeFSL = 3,
-  /** Will ignore orientation code and respect negative pixel dimensions */
-  AnalyzeSPM = 2,
-  /** Same as AnalyzeITK4 but will show warning about deprecated file format (Default)*/
-  AnalyzeITK4Warning = 1,
-  /** Reject Analyze files as potentially wrong  */
-  AnalyzeReject = 0
+public:
+  /** \class Analyze75Flavor
+   * \ingroup ITKIONIFTI
+   * Enum used to define the way to treat legacy Analyze75 files.
+   */
+  enum class Analyze75Flavor : uint8_t
+  {
+    /** Behavior introduced in ITK4.0 by NIFTI reader interpreting Analyze files */
+    AnalyzeITK4 = 4,
+    /** Will ignore orientation code and negative pixel dimensions */
+    AnalyzeFSL = 3,
+    /** Will ignore orientation code and respect negative pixel dimensions */
+    AnalyzeSPM = 2,
+    /** Same as AnalyzeITK4 but will show warning about deprecated file format (Default)*/
+    AnalyzeITK4Warning = 1,
+    /** Reject Analyze files as potentially wrong  */
+    AnalyzeReject = 0
+  };
+
+  /** \class NiftiFileEnum
+   * \ingroup ITKIONIFTI
+   * Enum used to define the possible file formats readable by this ImageIO implementation.
+   */
+  enum class NiftiFileEnum : int8_t
+  {
+    /** 2-file Nifti (consisting of .hdr and .img file). */
+    TwoFileNifti = 2,
+    /** 1-file Nifti (consisting of .nii file). */
+    OneFileNifti = 1,
+    /** Legacy Analyze 7.5 format (consisting of .hdr and .img file). */
+    Analyze75 = 0,
+    /** Some other file format, or file system error. */
+    OtherOrError = -1,
+  };
 };
```

Comment:
```suggestion
};

#if !defined(ITK_LEGACY_REMOVE)
  using Analyze75Flavor = NiftiImageIOEnums::Analyze75Flavor;
#endif
```

Reply:
Added in 807dcef. Thanks for the suggestions @dzenanz.